dotCMS/src/main/java/com/dotcms/security/apps/AppDescriptorImpl.java

```diff
@@ -28,13 +31,15 @@
      * @param allowExtraParameters
      */
     public AppDescriptorImpl(final String fileName,
+            final boolean systemApp,
             final String name,
             final String description,
             final String iconUrl,
             final Boolean allowExtraParameters,
             final Map<String, ParamDescriptor> params) {
         super(name, description, iconUrl, allowExtraParameters, params);
```

Comment:
Lots of parameters in the constructors - looks like a good candidate for a Builder pattern

Reply:
This constructor is public for the sake of being used on the DataGen (which is really a builder)
The real constructor is this:
` AppDescriptorImpl(final String fileName, final boolean systemApp, final AppSchema appSchema)`
It only takes 3 params. I am marking this constructor as @VivisbleForTesting